hist/hist/test/TFormulaHessianTests.cxx

```diff
@@ -0,0 +1,86 @@
+/// \file TFormulaHessianTests.cxx
+///
+/// \brief The file contain unit tests which test the clad-based hessian
+///        computations.
+///
+/// \author Baidyanath Kundu <kundubaidya99@gmail.com>
+///
+/// \date Aug, 2021
+///
+/*************************************************************************
+ * Copyright (C) 1995-2018, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#include "ROOTUnitTestSupport.h"
+
+#include <Math/MinimizerOptions.h>
+#include <TFormula.h>
+
+TEST(TFormulaHessianPar, Sanity)
+{
+   TFormula f("f", "x*std::sin([0]) - y*std::cos([1])");
+   double p[] = {30, 60};
+   f.SetParameters(p);
+   double x[] = {1, 2};
+   TFormula::CladStorage result(4);
+   f.HessianPar(x, result);
+
+   ASSERT_FLOAT_EQ(-1 * x[0] * std::sin(p[0]), result[0]);
+   ASSERT_FLOAT_EQ(0, result[1]);
+   ASSERT_FLOAT_EQ(0, result[2]);
+   ASSERT_FLOAT_EQ(x[1] * std::cos(p[1]), result[3]);
+}
+
+TEST(TFormulaHessianPar, ResultUpsize)
+{
+   TFormula f("f", "std::sin([1]) - std::cos([0])");
+   double p[] = {60, 30};
+   f.SetParameters(p);
+   TFormula::CladStorage result;
+   double x[] = {2, 1};
+
+   ASSERT_TRUE(0 == result.size());
+   ROOT_EXPECT_WARNING(f.HessianPar(x, result),
+   "TFormula::HessianPar",
+   "The size of hessian result is 0 but 4 is required. Resizing."
+   );
+
+   ASSERT_FLOAT_EQ(std::cos(p[0]), result[0]);
+   ASSERT_FLOAT_EQ(0, result[1]);
+   ASSERT_FLOAT_EQ(0, result[2]);
+   ASSERT_FLOAT_EQ(- std::sin(p[1]), result[3]);
+   ASSERT_TRUE(4 == result.size());
+}
+
+TEST(TFormulaHessianPar, ResultDownsize)
+{
+   TFormula f("f", "std::sin([0])");
+   double p[] = {60};
+   f.SetParameters(p);
+   TFormula::CladStorage result(2);
```

Comment:
I don't understand this test. Why you are creating a result of size(2) when it should be 1 ?

Reply:
It's motive is to check whether larger sized vectors work or do they cause errors. A similar test is present in `TFormulaGradientTests`. Although it's not very robust.